### PR TITLE
Menu editor xtd

### PR DESCRIPTION
**Pull request by infograf768**
Grace to the new modals for menu items, we can now add a new editor-xtd to insert menu items links in content.

Instructions:
Install a 3.7.0 (not necessary to patch first with https://github.com/joomla/joomla-cms/pull/11694 but modal title would look better)
Patch with this PR.
Discover the new plugin. Enable it.

Edit an article or create a new one
The editor now shows a new button "Menu"

screen shot 2016-08-22 at 10 33 48

depending if the site is multilingual or not, the language will be appended to the link when the menu item is tagged to a specific content language.

By toggling the editor you should get:
`<p><a href="index.php?Itemid=111">newsfeed category menu</a></p>`
or
`<p><a href="index.php?Itemid=107&amp;lang=fr-FR">menu article fr-FR</a></p>`

test the results in frontend.

### CODE REVIEW

**Comment by brianteeman** [comment]
The PR says it is for 3.7 but you have done a PR against the staging branch. This has resulted in 199 changed files the majority of which are nothing to do with this PR

File: plugins/editors-xtd/menu/menu.php [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+ * @copyright   Copyright (C) 2005 - 2016 Open Source Matters, Inc. All rights reserved.
+ * @license     GNU General Public License version 2 or later; see LICENSE.txt
+ */
+
+defined('_JEXEC') or die;
+
+/**
+ * Editor menu buton
+ *
+ * @since  3.7
```
Comment (brianteeman) [inline plugins/editors-xtd/menu/menu.php]: can you make this
 * @since  __DEPLOY_VERSION__

File: plugins/editors-xtd/menu/menu.php [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+ */
+class PlgButtonMenu extends JPlugin
+{
+	/**
+	 * Load the language file on instantiation.
+	 *
+	 * @var    boolean
+	 * @since  3.7
```
Comment (brianteeman) [inline plugins/editors-xtd/menu/menu.php]: can you make this

@since DEPLOY_VERSION

**Comment by infograf768** [comment]
closed as targetted to wrong branch<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/11720">issues.joomla.org/joomla-cms/11720</a>.</sub>

File: administrator/language/en-GB/en-GB.plg_editors-xtd_menu.sys.ini [hunk 3]
```diff
@@ -0,0 +1,7 @@
+; Joomla! Project
+; Copyright (C) 2005 - 2016 Open Source Matters. All rights reserved.
+; License GNU General Public License version 2 or later; see LICENSE.txt, see LICENSE.php
+; Note : All ini files need to be saved as UTF-8
+
+PLG_EDITORS-XTD_MENU="Button - Menu"
+PLG_MENU_XML_DESCRIPTION="Displays a button to make it possible to insert menu items links into an Article. Displays a popup allowing you to choose the menu item."
```
Comment (brianteeman) [inline administrator/language/en-GB/en-GB.plg_editors-xtd_menu.sys.ini]: this should be "menu item links"

File: administrator/language/en-GB/en-GB.plg_editors-xtd_menu.ini [hunk 4]
```diff
@@ -0,0 +1,8 @@
+; Joomla! Project
+; Copyright (C) 2005 - 2016 Open Source Matters. All rights reserved.
+; License GNU General Public License version 2 or later; see LICENSE.txt, see LICENSE.php
+; Note : All ini files need to be saved as UTF-8
+
+PLG_EDITORS-XTD_MENU="Button - Menu"
+PLG_MENU_BUTTON_MENU="Menu"
+PLG_MENU_XML_DESCRIPTION="Displays a button to make it possible to insert menu items links into an Article. Displays a popup allowing you to choose the menu item."
```
Comment (brianteeman) [inline administrator/language/en-GB/en-GB.plg_editors-xtd_menu.ini]: this should be "menu item links"

File: plugins/editors-xtd/menu/menu.php [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+		 * Currently uses blank class.
+		 */
+		$link = 'index.php?option=com_menus&amp;view=items&amp;layout=modal&amp;tmpl=component&amp;' . JSession::getFormToken() . '=1';
+
+		$button          = new JObject;
+		$button->modal   = true;
+		$button->class   = 'btn';
+		$button->link    = $link;
+		$button->text    = JText::_('PLG_MENU_BUTTON_MENU');
+		$button->name    = 'file-add';
```
Comment (brianteeman) [inline plugins/editors-xtd/menu/menu.php]: Can we use a different icon please so we dont have the same icon for two different things. (file-add is also used for modules)

The list of available icons is here https://docs.joomla.org/J3.x:Joomla_Standard_Icomoon_Fonts

**Pull request closed without merging**